### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by lokesh-vr-17773**
**Issue**
When WhisperX is installed with Poetry, no dependencies are being installed.

**Change**
Converted the [setup.py](http://setup.py/) and requirements.txt files to pyproject.toml (using setuptools as the backend).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+[project]
+name        = "whisperx"
+description = "Time-Accurate Automatic Speech Recognition using Whisper."
+version     = "3.1.1"
+readme      = "README.md"
+requires-python = ">=3.8"
+authors = [
+  { name= "Max Bain" },
+]
+license = { text = "MIT" }
```
Comment (sorgfresser) [inline pyproject.toml:14]: The license is BSD.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+dev = [ "pytest" ]
+
+[tool.setuptools]
+py-modules = [ "whisperx" ]
+packages = [ "whisperx" ]
+include-package-data = true
+license-files = [ "LICENSE" ]
+
+[tools.setuptools.packages.find]
+exclude = [ "tests*" ]
```
Comment (sorgfresser) [inline pyproject.toml:43]: again, there are no tests (yet).

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+]
+
+[project.scripts]
+whisperx = "whisperx.transcribe:cli"
+
+[project.urls]
+"Homepage" =  "https://github.com/m-bain/whisperx"
+
+[project.optional-dependencies]
+dev = [ "pytest" ]
```
Comment (sorgfresser) [inline pyproject.toml:34]: Is pytest a thing used somewhere? I know you took it from setup.py, but still.

**Comment by adrianlzt** [comment]
Any update on this?

The issue with poetry is because the install_requires is a list comprehension.
Poetry issue: https://github.com/python-poetry/poetry/issues/8774

**Comment by Barabazs** [comment]
This is a Poetry issue and not whisperx. Closing this for this reason and due to being incomplete and abandoned.

**Pull request closed without merging**